src/main/java/org/datadog/jenkins/plugins/datadog/util/git/RepositoryInfoCallback.java

```diff
@@ -24,23 +24,25 @@
     @Override
     public RepositoryInfo invoke(Repository repository, VirtualChannel channel) throws IOException, InterruptedException {
         try {
+            System.out.println("Repository: " + repository);
             Ref head = repository.getRefDatabase().findRef("HEAD");
             if(head == null) {
-                LOGGER.fine("Unable to build RepositoryInfo. HEAD is null.");
-                return null;
+                LOGGER.info("Unable to build RepositoryInfo. HEAD is null.");
+                return RepositoryInfo.EMPTY_REPOSITORY_INFO;
             }
 
             // Discarded if it's not a symbolic to refs.
             if(!head.isSymbolic()) {
-                LOGGER.fine("Unable to build RepositoryInfo. HEAD is not symbolic.");
-                return null;
+                LOGGER.info("Unable to build RepositoryInfo. HEAD is not symbolic.");
+                LOGGER.info("HEAD: " + head);
+                return RepositoryInfo.EMPTY_REPOSITORY_INFO;
             }
 
             final String defaultBranch = GitInfoUtils.normalizeBranch(head.getTarget().getName());
             return new RepositoryInfo(defaultBranch);
 
         } catch (Exception e) {
-            LOGGER.fine("Unable to build RepositoryInfo. Error: " + e);
+            LOGGER.info("Unable to build RepositoryInfo. Error: " + e);
             return null;
```

Comment:
Why aren't you returning RepositoryInfo.EMPTY_REPOSITORY_INFO here?

Reply:
For errors, we return null to trigger again the logic in case the repository info was calculated again. By other hand, the case where we actually know that the info could be calculated but there is no valid info for us, we return the empty repository info.